backend/code/m-booking/src/test/java/com/mbooking/service/ReservationServiceIntegrationTests.java

```diff
@@ -0,0 +1,370 @@
+package com.mbooking.service;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.mbooking.dto.MakeReservationResponseDTO;
+import com.mbooking.dto.ReservationDTO;
+import com.mbooking.dto.ReservationDetailsDTO;
+import com.mbooking.exception.DuplicateSeatsException;
+import com.mbooking.exception.EmptyReservationDetailsException;
+import com.mbooking.exception.ManifestationReservationsAvailableException;
+import com.mbooking.exception.MaxReservationsException;
+import com.mbooking.exception.NoMoreSpaceException;
+import com.mbooking.exception.NoSuchManifestationDayException;
+import com.mbooking.exception.NoSuchManifestationException;
+import com.mbooking.exception.NoSuchSeatException;
+import com.mbooking.exception.NoSuchSectionException;
+import com.mbooking.exception.NoSuchUserException;
+import com.mbooking.exception.ReservableUntilException;
+import com.mbooking.exception.SeatTakenException;
+import com.mbooking.exception.SectionNotFromSameManifestationException;
+import com.mbooking.model.Reservation;
+import com.mbooking.repository.ReservationRepository;
+import com.mbooking.repository.UserRepository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
+@TestPropertySource(locations="classpath:application-test_reservation.properties")
+public class ReservationServiceIntegrationTests {
```

Comment:
Since we are in the test class, I don't think it was necessary to add 'test' prefix to all method names. On the other hand, it won't do any harm